Create a Python virtual environment (#6681)
fishtown-analytics/docs.getdbt.com: website/docs/docs/core/create-a-python-virtual-environment.md

---

> title: Create a Python Virtual Environment

Reviewer: our [style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md#titles) mentions we should use sentence case, not title case
```suggestion
title: Create a Python virtual environment
```

The Python ecosystem offers several tools for creating isolated environments, such as [conda](https://anaconda.org/anaconda/conda), [poetry](https://python-poetry.org/docs/managing-environments/), and `venv`. Among these, `venv` has the fewest additional dependencies and has been included by default in recent Python versions for quite some time.

`venv` will set up a Python virtual environment within the `.venv` folder.

Users who want to run dbt locally, for example in [dbt Core](/docs/core/installation-overview) or the [dbt Cloud CLI](/docs/cloud/cloud-cli-installation#install-a-virtual-environment) may want to install a Python virtual environment. 

## Prerequisites

- Have [Python](https://www.python.org/downloads/) installed on your machine. You can check if Python is installed by running `python --version` or `python3 --version` in your terminal or command prompt.
- Have [pip installed](https://pip.pypa.io/en/stable/installation/). You can check if pip is installed by running `pip --version` or `pip3 --version`.

> - Access to a terminal or command prompt.

Reviewer: sugg moving 'access to terminal first' before we introduce the python and pip install requirements?

- Have the necessary permissions to create directories and install packages on your machine.

## Install a Python virtual environment 

Depending on the operating system you use, you'll need to execute specific steps to set up a virtual environment. 

To install a Python virtual environment, navigate to your project directory and execute the command. This will generate a new virtual environment within a local folder named `venv`:

> <Expandable alt_header="Unix/macOS" >

Reviewer: looks great! i wonder if using tabs is better for users though so its less clicking

1. Create your virtual environment

```shell
python3 -m venv .venv
```

2. Activate your virtual environment:

```shell
source .venv/bin/activate
which python
.venv/bin/python
```
</Expandable>

<Expandable alt_header="Windows" >

1. Create your virtual environment

```shell
py -m venv .venv
```

2. Activate your virtual environment:

```shell
.venv\Scripts\activate
where python
.venv\Scripts\python
```

</Expandable>

> 

Reviewer: what's the next step after creating a virtual environment? is it then to install dbt via pip? if so, might be good to have a link or next steps section telling users what to do next?

---
title: Create a Python virtual environment
id: create-a-python-virtual-environment
description: "Instructions on creating a Python virtual environment."
pagination_next: "docs/core/create-a-python-virtual-environment"
pagination_prev: null
---

A Python virtual environment is an isolated workspace for Python projects. This prevents libraries and versions used in one project from interfering with others, making it especially helpful when working on multiple projects with differing requirements or avoiding conflicts with global Python installations.

The Python ecosystem offers several tools for creating isolated environments, such as [conda](https://anaconda.org/anaconda/conda), [poetry](https://python-poetry.org/docs/managing-environments/), and `venv`. Among these, `venv` has the fewest additional dependencies and has been included by default in recent Python versions for quite some time.

> `venv` will set up a Python virtual environment within the `.venv` folder.

Author: ```suggestion
`venv` will set up a Python virtual environment within the `env` folder.
```

Users who want to run dbt locally, for example in [dbt Core](/docs/core/installation-overview) or the [dbt Cloud CLI](/docs/cloud/cloud-cli-installation#install-a-virtual-environment) may want to install a Python virtual environment. 

## Prerequisites

- Access to a terminal or command prompt.
- Have [Python](https://www.python.org/downloads/) installed on your machine. You can check if Python is installed by running `python --version` or `python3 --version` in your terminal or command prompt.
- Have [pip installed](https://pip.pypa.io/en/stable/installation/). You can check if pip is installed by running `pip --version` or `pip3 --version`.
- Have the necessary permissions to create directories and install packages on your machine.

## Install a Python virtual environment 

Depending on the operating system you use, you'll need to execute specific steps to set up a virtual environment. 

> To install a Python virtual environment, navigate to your project directory and execute the command. This will generate a new virtual environment within a local folder named `venv`:

Author: ```suggestion
To install a Python virtual environment, navigate to your project directory and execute the command. This will generate a new virtual environment within a local folder which you can name it anything you want.  [Our convention](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#virtual-environments) has been to name it `env` or `env-anything-you-want`
```

<Tabs>
  <TabItem value="Unix/macOS" label="Unix/macOS">
    1. Create your virtual environment

    ```shell

>     python3 -m venv .venv

Author: ```suggestion
    python3 -m venv env
```

    ```

    2. Activate your virtual environment:

    ```shell

>     source .venv/bin/activate

Author: ```suggestion
    source env/bin/activate
```

    which python

>     .venv/bin/python

Author: ```suggestion
    env/bin/python
```

    ```
  </TabItem>

  <TabItem value="Windows" label="Windows">
    1. Create your virtual environment

    ```shell

>     py -m venv .venv

Author: ```suggestion
    py -m venv env
```

    ```

    2. Activate your virtual environment:

    ```shell

>     .venv\Scripts\activate

Author: ```suggestion
    env\Scripts\activate
```

    where python

>     .venv\Scripts\python

Author: ```suggestion
    env\Scripts\python
```

    ```
  </TabItem>

> </Tabs>

Reviewer 2: <img width="530" alt="Screenshot 2024-12-25 at 4 27 37 PM" src="https://github.com/user-attachments/assets/305d06bf-055e-44b5-867d-692ebab6d3cf" />

Can be render it better?

`venv` will set up a Python virtual environment within the `env` folder.

Users who want to run dbt locally, for example in [dbt Core](/docs/core/installation-overview) or the [dbt Cloud CLI](/docs/cloud/cloud-cli-installation#install-a-virtual-environment) may want to install a Python virtual environment. 

## Prerequisites

- Access to a terminal or command prompt.
- Have [Python](https://www.python.org/downloads/) installed on your machine. You can check if Python is installed by running `python --version` or `python3 --version` in your terminal or command prompt.
- Have [pip installed](https://pip.pypa.io/en/stable/installation/). You can check if pip is installed by running `pip --version` or `pip3 --version`.
- Have the necessary permissions to create directories and install packages on your machine.

## Install a Python virtual environment 

Depending on the operating system you use, you'll need to execute specific steps to set up a virtual environment. 

> To install a Python virtual environment, navigate to your project directory and execute the command. This will generate a new virtual environment within a local folder which you can name it anything you want.  [Our convention](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#virtual-environments) has been to name it `env` or `env-anything-you-want`

Reviewer 2: `which you can name it anything you want` -> `that you can name anything`.
A suggestion on refining the literature.

> To install a Python virtual environment, navigate to your project directory and execute the command. This will generate a new virtual environment within a local folder which you can name it anything you want.  [Our convention](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#virtual-environments) has been to name it `env` or `env-anything-you-want`

Author: ```suggestion
To install a Python virtual environment, navigate to your project directory and execute the command. This will generate a new virtual environment within a local folder that you can name anything.  [Our convention](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#virtual-environments) has been to name it `env` or `env-anything-you-want`
```
